dotnet/src/Agents/Abstractions/Definition/AgentDefinition.cs

```diff
@@ -14,171 +14,60 @@ public sealed class AgentDefinition
     /// </summary>
-    public string? Type
-    {
-        get => this._type;
-        set
-        {
-            Verify.NotNull(value);
-            this._type = value;
-        }
-    }
+    public string? Type { get; set; }
 
     /// <summary>
     /// Gets or sets the name of the  agent.
     /// </summary>
-    public string? Name
-    {
-        get => this._name;
-        set
-        {
-            Verify.NotNull(value);
-            this._name = value;
-        }
-    }
+    public string? Name { get; set; }
 
     /// <summary>
     /// Gets or sets the description of the agent.
     /// </summary>
-    public string? Description
-    {
-        get => this._description;
-        set
-        {
-            Verify.NotNull(value);
-            this._description = value;
-        }
-    }
+    public string? Description { get; set; }
 
     /// <summary>
     /// Gets or sets the instructions for the agent to use.
     /// </summary>
-    public string? Instructions
-    {
-        get => this._instructions;
-        set
-        {
-            Verify.NotNull(value);
-            this._instructions = value;
-        }
-    }
+    public string? Instructions { get; set; }
 
     /// <summary>
     /// Gets or sets the metadata associated with the agent.
     /// </summary>
-    public IDictionary<string, object?>? Metadata
-    {
-        get => this._metadata;
-        set
-        {
-            Verify.NotNull(value);
-            this._metadata = value;
-        }
-    }
+    public IDictionary<string, object?>? Metadata { get; set; }
 
     /// <summary>
     /// Gets or sets the model used by the agent.
     /// </summary>
-    public ModelDefinition? Model
-    {
-        get => this._model;
-        set
-        {
-            Verify.NotNull(value);
-            this._model = value;
-        }
-    }
+    public ModelDefinition? Model { get; set; }
 
     /// <summary>
     /// Gets or sets the collection of input variables used by the agent.
     /// </summary>
-    public IList<InputVariable> Inputs
-    {
-        get => this._inputs ??= [];
-        set
-        {
-            Verify.NotNull(value);
-            this._inputs = value;
-        }
-    }
+    public IList<InputVariable>? Inputs { get; set; }
 
     /// <summary>
     /// Gets or sets the collection of output variables supported by the agent.
     /// </summary>
-    public IList<OutputVariable> Outputs
-    {
-        get => this._outputs ??= [];
-        set
-        {
-            Verify.NotNull(value);
-            this._outputs = value;
-        }
-    }
+    public IList<OutputVariable>? Outputs { get; set; }
 
     /// <summary>
     /// Gets or sets the template options used by the agent.
     /// </summary>
-    public TemplateOptions? Template
-    {
-        get => this._template;
-        set
-        {
-            Verify.NotNull(value);
-            this._template = value;
-        }
-    }
+    public TemplateOptions? Template { get; set; }
```

Comment:
"Template" naming here can give multiple understanding, might better add the `Options` suffix as well.

```suggestion
    public TemplateOptions? TemplateOptions { get; set; }
```

Reply:
The naming is being decided by the WG for the agent declarative format and this is the current proposal.